debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/ModularityHelper.java

```diff
@@ -340,4 +342,18 @@ public static int iterateClassLoaderClasses(PrintStream out, ClassIteratorFilter
 		return count;
 	}
 
+	public static String getModuleName(J9ModulePointer modulePtr) throws CorruptDataException {
+	    String moduleName = null;
+	    if (modulePtr != null) {
+	        Object mn = modulePtr.moduleName();
+	        if (mn instanceof J9ObjectPointer) {
+	            moduleName = J9ObjectHelper.stringValue((J9ObjectPointer)mn);
+	        } else if (mn instanceof J9UTF8Pointer) {
+	            moduleName = J9UTF8Helper.stringValue((J9UTF8Pointer)mn);
+	        } else {
+	            throw new CorruptDataException("Invalid module name data type encountered: " + mn.toString());
+	        }
+	    }
+	    return moduleName;
```

Comment:
We might get away with using one name by adding a new constant, with a default representing the old type, in `CompatibilityConstants29.dat` by using `moduleNameEA()` which should consistently return `PointerPointer` for both cases. That address can be conditionally cast by testing that new constant, by using either `J9ObjectPointer.cast()` or `J9UTF8Pointer.cast()` as appropriate.

Reply:
I am testing a patch now that uses `moduleNameEA()` in a similar manner (without the considering `CompatibilityConstants29.dat`).

Reply:
As is, applied to a core produced by an older JVM this is what happens:
```
> !findallmodules
Error executing DDR command !findallmodules : null
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
	at java.base/java.lang.reflect.Method.invoke(Method.java:586)
	at openj9.dtfjview/com.ibm.jvm.dtfjview.CombinedContext.executeDDRInteractiveCommand(CombinedContext.java:258)
	at openj9.dtfjview/com.ibm.jvm.dtfjview.CombinedContext.execute(CombinedContext.java:169)
	at openj9.dtfjview/com.ibm.jvm.dtfjview.Session.execute(Session.java:813)
	at openj9.dtfjview/com.ibm.jvm.dtfjview.Session.execute(Session.java:767)
	at openj9.dtfjview/com.ibm.jvm.dtfjview.tools.ToolsRegistry.executeJdmpviewCommand(ToolsRegistry.java:183)
	at openj9.dtfjview/com.ibm.jvm.dtfjview.tools.ToolsRegistry.execute(ToolsRegistry.java:107)
	at openj9.dtfjview/com.ibm.jvm.dtfjview.tools.ToolsRegistry.execute(ToolsRegistry.java:88)
	at openj9.dtfjview/com.ibm.jvm.dtfjview.tools.ToolsRegistry.recordAndExecute(ToolsRegistry.java:72)
	at openj9.dtfjview/com.ibm.jvm.dtfjview.Session.runInteractive(Session.java:741)
	at openj9.dtfjview/com.ibm.jvm.dtfjview.Session.run(Session.java:665)
	at openj9.dtfjview/com.ibm.jvm.dtfjview.DTFJView.launch(DTFJView.java:52)
	at openj9.dtfjview/com.ibm.jvm.dtfjview.DTFJView.main(DTFJView.java:46)
Caused by: java.lang.NoSuchMethodError: com/ibm/j9ddr/vm29/pointer/generated/J9ModulePointer.moduleName()Lcom/ibm/j9ddr/vm29/pointer/generated/J9UTF8Pointer; (loaded from <Unknown> by J9DDRClassloader for com.ibm.j9ddr.vm29.) called from class com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper (loaded from <Unknown> by J9DDRClassloader for com.ibm.j9ddr.vm29.).
	at com.ibm.j9ddr.vm29.tools.ddrinteractive.ModularityHelper.getModuleName(ModularityHelper.java:348)
	at com.ibm.j9ddr.vm29.tools.ddrinteractive.commands.FindAllModulesCommand.run(FindAllModulesCommand.java:73)
	at openj9.dtfj/com.ibm.j9ddr.tools.ddrinteractive.Context.tryCommand(Context.java:229)
	at openj9.dtfj/com.ibm.j9ddr.tools.ddrinteractive.Context.execute(Context.java:202)
	at openj9.dtfj/com.ibm.j9ddr.tools.ddrinteractive.DDRInteractive.execute(DDRInteractive.java:356)
	at openj9.dtfj/com.ibm.j9ddr.command.CommandReader.processLine(CommandReader.java:78)
	at openj9.dtfj/com.ibm.j9ddr.tools.ddrinteractive.DDRInteractive.processLine(DDRInteractive.java:331)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	... 13 more
```

Reply:
Yes, same as mentioned in https://github.com/eclipse-openj9/openj9/pull/20740#discussion_r1876552305

Reply:
> I am testing a patch now that uses `moduleNameEA()` in a similar manner (without the considering `CompatibilityConstants29.dat`).

Please explain how you intend to decide which way to cast without a new constant (and help from `CompatibilityConstants29.dat`).

Reply:
I was catching the `NoSuchMethodError` shown above, and handling the `J9Object*` version in the `catch` block. We should add the constant though. I don't like relying on try-catch though; I wasn't aware of `CompatibilityConstants29.dat` before you mentioned it

Reply:
I'm not a fan of the `Ptr` suffix for the parameter name (you wouldn't use it in C code); suggest removing it.

The null check needs to be expanded:
```java
		if ((module != null) && module.notNull()) {
```

Reply:
For this, are we OK to do something like:

```
	public static String getModuleName(J9ModulePointer module) throws CorruptDataException {
		String moduleName = null;
		if ((module != null) && module.notNull()) {
			try {
				J9UTF8Pointer mn = module.moduleName();
				moduleName = J9UTF8Helper.stringValue(mn);
			} catch (NoSuchMethodError e) {
				/* moduleName used to be a J9ObjectPointer. */
				J9ObjectPointer mn = J9ObjectPointer.cast(module.moduleNameEA().at(0));
				moduleName = J9ObjectHelper.stringValue(mn);
			}
		}
		return moduleName;
	}
```

Or would you rather still explore a version involving a constant?

Reply:
Actually, I've added the constant: `J9Consts.J9_MODULE_NAME_IS_J9UTF8` which is `0` in `CompatibilityConstants29.dat` and `0x1` in `j9consts.h`. And changed this function to:
```
public static String getModuleName(J9ModulePointer module) throws CorruptDataException {
		String moduleName = null;
		if ((module != null) && module.notNull()) {
			if (0 != J9Consts.J9_MODULE_NAME_IS_J9UTF8) {
				J9UTF8Pointer mn = module.moduleName();
				moduleName = J9UTF8Helper.stringValue(mn);
			} else {
				/* moduleName used to be a J9ObjectPointer. */
				J9ObjectPointer mn = J9ObjectPointer.cast(module.moduleNameEA().at(0));
				moduleName = J9ObjectHelper.stringValue(mn);
			}
		}
		return moduleName;
	}
```

I like this more.